### PR TITLE
(maint) Improve usage of ActionRequest in RequestProcessor

### PR DESCRIPTION
**Pull request by parisiale**
- adding results_dir_ to ActionRequest;
- tidying up RequestProcessor and ResultsStorage.

### CODE REVIEW

File: lib/src/request_processor.cc [hunk 1]
```diff
@@ -133,12 +133,12 @@ void nonBlockingActionTask(std::shared_ptr<Module> module_ptr,
 
     try {
         ResultsMutex::LockGuard a_l { ResultsMutex::Instance().access_mtx };
-        mtx_ptr = ResultsMutex::Instance().get(transaction_id);
+        mtx_ptr = ResultsMutex::Instance().get(request.transactionId());
```
Comment (mruzicka) [inline lib/src/request_processor.cc]: should the change from ```transaction_id``` to ```request.transactionId()``` go into a separate commit?

**Comment by mruzicka** [comment]
@parisiale except for the single comment this looks good to me. If you think another commit is not necessary, I'm happy to merge as is.

File: lib/src/request_processor.cc (hunk 1 above)
Comment (parisiale) [inline lib/src/request_processor.cc]: No. This is about "Rely on ActionRequest".
